Sources/GRPCCore/Call/Client/Internal/ClientStreamExecutor.swift

```diff
@@ -96,7 +96,11 @@ internal enum ClientStreamExecutor {
       try await stream.write(.metadata(request.metadata))
       try await request.producer(.map(into: stream) { .message(try serializer.serialize($0)) })
     }.castError(to: RPCError.self) { other in
-      RPCError(code: .unknown, message: "Write failed.", cause: other)
+      if let convertible = other as? any RPCErrorConvertible {
+        RPCError(convertible)
```

Comment:
It's probably worth pushing this logic into `castError` (or a new equivalent helper method) since we do this in a bunch of places now.

Reply:
I introduced a `castOrConvertRPCError` helper to extract this 👍 I think there was one additional spot in the project were a similar conversion was made and replaced that as well.